client/chat/connected-channel-name.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import styled from 'styled-components'
+import { SbChannelId } from '../../common/chat'
+import { useAppSelector } from '../redux-hooks'
+import { colorDividers } from '../styles/colors'
+
+const LoadingName = styled.span`
+  margin-right: 0.25em;
+  background-color: ${colorDividers};
+  border-radius: 2px;
+`
+
+export interface ConnectedChannelNameProps {
+  className?: string
+  /** The channel to show the corresponding name for. */
+  channelId: SbChannelId
+}
+
+/**
+ * A component which, given a channel ID, displays a channel name. Pretty barebones at the moment,
+ * but will be extended with further functionality soon.
+ */
+export function ConnectedChannelName({ className, channelId }: ConnectedChannelNameProps) {
+  const channel = useAppSelector(s => s.chat.byId.get(channelId))
```

Comment:
This gets a bit weird, because potentially someone might want to use this component to display a channel name for a channel that someone hasn't joined, but you've split the reducers for joined channels and unjoined ones. I think a better structure would be to make a reducer for messages/user lists, and one for storing information about the channel (name, etc.), or make a single reducer that combines all of those things. But having 2 reducers that can both feasibly be used to look up a channel name is gonna lead to a lot of weird bugs. In general try to only have *1* place to store and look up the same information

Reply:
It also isn't really safe imo to display this information without also triggering retrieval of it if it doesn't exist, should ideally do stuff similar to what we do for users with microtasks + batch retrieval. (Not very nice to users to tell them we're loading something that we never make an attempt to load :) )

Reply:
I left the second part of your comment (triggering the retrieval of channel info) for a next PR if you don't mind. I only made a barebones version of this component for, but I will expand it significantly once I add channel mentions, which should be soon.

Reply:
You should file an issue for it, but sure